### PR TITLE
[HttpKernel] Let the storage manage the session starts

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24730
| License       | MIT
| Doc PR        | ø

HttpKernel's request collector should not really care if the session has started or not, be let the storage decide. Without the session, it is not possible to track the redirected pages. 

I don't think the consideration of "we should not start the session if not needed by the user's code" applies here as if this is running, that is very likely that the user is running the dev environment anyway.

### CODE REVIEW

**Comment by mvrhov** [comment]
IMO that's not correct fix. Data collector should not start the session.

**Comment by sroze** [comment]
@mvrhov hehe, hence my last sentence 😉 Do you have a better option to fix #24730 ?

**Comment by mvrhov** [comment]
Not, from the top of my head. But I know that I fixed something similar e.g DataCollector starting session somewhere in sf 2.0 times. As it was a great annoyance.

**Comment by stof** [comment]
@nicolas-grekas can you check this to see what is the impact of the changes you made to the session and how this should best be solved ?

**Comment by nicolas-grekas** [comment]
The impact is meeting the expectations set then, nothing more: if the session is empty, it isn't written nor started. This "isStarted" check is strange, as it binds a feature of the profiler to the behavior of the page being monitored. Maybe legitimate of course. This PR looks ok to me thought.

**Comment by sroze** [comment]
Removed the test asserting that the session is not started, as obviously, this might be now...

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @sroze.

**Comment by kbond** [comment]
This change is causing a functional test of mine to fail. I am testing that deleting the current session logs the user out. I am calling `$client->getCookieJar()->->expire('MOCKSESSID');`, then visiting another page - the user is still logged in. Test environment using `MockFileSessionStorage` only.

**Comment by sroze** [comment]
@kbond could you open an issue for this, explaining exactly what is wrong? As I'm not sure how this change would impact your test. Cheers!

File: src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php [hunk 1]
```diff
@@ -315,7 +315,7 @@ public function onKernelController(FilterControllerEvent $event)
 
     public function onKernelResponse(FilterResponseEvent $event)
     {
-        if (!$event->isMasterRequest() || !$event->getRequest()->hasSession() || !$event->getRequest()->getSession()->isStarted()) {
+        if (!$event->isMasterRequest() || !$event->getRequest()->hasSession()) {
```
Comment (kbond) [inline src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php]: Specifically, this change is causing my problem.

**Pull request closed without merging**